stdlib/LinearAlgebra/test/dense.jl

```diff
@@ -395,6 +395,19 @@ end
     @test kron(b',2) == [8 10 12]
 end
 
+@testset "kron!" begin
+    a = [1.0, 0.0]
+    b = [0.0, 1.0]
+    @test kron!([1.0, 0.0], b, 0.5) == [0.0; 0.5]
```

Comment:
I'm confused we don't have a
```julia
kron!(c::AbstractVecOrMat, a:: Number, b::AbstractVecOrMat) = mul!(c, a, b)
```
method. It wouldn't get called by `kron(a::Number, b::AbstractVecOrMat)`, but the method you're testing here doesn't get called by `kron` either. Maybe this would be an opportunity to add that simple method and add a test with flipped arguments here?